cni/network/invoker_cns.go

```diff
@@ -175,9 +175,16 @@ func (invoker *CNSIPAMInvoker) Add(addConfig IPAMAddConfig) (IPAMAddResult, erro
 				return IPAMAddResult{}, err
 			}
 		default:
+			// Assign default index
+			if defaultIndex == -1 {
+				// add a default Interface
+				// would put within configureDefaultAddResult() but we have a dependency on numInterfacesWithDefaultRoutes
+				addResult.interfaceInfo = append(addResult.interfaceInfo, network.InterfaceInfo{NICType: cns.InfraNIC})
+				defaultIndex = len(addResult.interfaceInfo) - 1
```

Comment:
Correct me if I am wrong,
The `addresult.interfaceInfo` will be empty if `defaultindex` is -1, so we append the result. 
This `default` case will be called multiple times or once ? (Asking because of the condition `deafultIndex == -1`
 Why don't we use the `findDefaultIndex` function here ?

Reply:
> Why don't we use the findDefaultIndex function here ? + The addresult.interfaceInfo will be empty if defaultindex is -1, so we append the result.

We could, but considering that `addResult.interFaceInfo[]` has not been assigned an InfraNIC we can get away with not using it. Saving some cycles of iterating through the array. The `-1` here is being used as a short lived bool.

I can't really think of another way to have an InfraNIC interface created only once, but not erroneously with only 1 variable.

> This default case will be called multiple times or once ? (Asking because of the condition deafultIndex == -1

Multiple.

Reply:
> I can't really think of another way to have an InfraNIC interface created only once, but not erroneously with only 1 variable.

Actually, this might be possible with a map as we get an err if the key does not exist, but as it stands, I don't think we have anything that we can use as a standard unique key throughout the codebase.